src/components/ReportActionItem/MoneyRequestView.tsx

```diff
@@ -170,6 +178,18 @@ function MoneyRequestView({
 
     let amountDescription = `${translate('iou.amount')}`;
 
+    const hasRoute = TransactionUtils.hasRoute(transaction, isDistanceRequest);
+    const rateID = transaction?.comment.customUnit?.customUnitRateID ?? '0';
+
+    const currency = policy ? policy.outputCurrency : PolicyUtils.getPersonalPolicy()?.outputCurrency ?? CONST.CURRENCY.USD;
+
+    const mileageRate = TransactionUtils.isCustomUnitRateIDForP2P(transaction) ? DistanceRequestUtils.getRateForP2P(currency) : rates[rateID as string] ?? {};
```

Comment:
NAB: There's another complication here that we forgot about. Let's handle it in a follow up PR please, since I don't want to delay this one more and add more complexity.

If we're viewing an old request, and the defaultP2P rates have changed since the request was created, then we might display the wrong rate because the rate the request was created with is no longer in the default mapping.

Therefore when viewing a created request we should first try to get the default rate from `transaction.comment.customUnit.defaultP2PRate`. It stores the rate value number only.

Reply:
@neil-marcellini ok I missed this one - will implement this